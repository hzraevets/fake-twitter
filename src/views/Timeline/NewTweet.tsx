import React, { useContext, useState } from 'react';
import { Card, Input, Button, Space } from 'antd';
import { useMutation } from 'react-query';

import { IdentityContext } from 'effects/Identity';
import { TweetContext } from 'effects/Tweet';
import {
  addNewTweetMutation,
  onAddNewTweetMutationSuccess,
} from 'query/TimeLine';

const { TextArea } = Input;

export function NewTweet() {
  const { loginUser, identity } = useContext(IdentityContext);
  const { addNewTweet } = useContext(TweetContext);
  const [ content, updateContent ] = useState('');
  const mutation = useMutation(addNewTweetMutation, { onSuccess: onAddNewTweetMutationSuccess });

  const handleTweet: React.MouseEventHandler = (event) => {
    event.preventDefault();

    const author = loginUser || 'UNKNOWN_AUTHOR';
    const firstname = loginUser ? identity[loginUser].firstname : 'Unknown';

    mutation.mutate({ author, firstname, content, addNewTweet });
    updateContent(''); // reset textarea
  }

  return (
    <Card className="w-80 break-inside-avoid-column mb-4">
      <Space direction="vertical" size="middle" className="w-full">
        <TextArea
          rows={3}
          className="max-h-32"
          placeholder="What's happening?"
          value={content}
          onChange={e => updateContent(e.target.value)}
        />
        <Button
          type="primary"
          className="float-right"
          disabled={content === '' || mutation.isLoading}
          onClick={handleTweet}
        >Tweet</Button>
      </Space>
    </Card>
  );
}
