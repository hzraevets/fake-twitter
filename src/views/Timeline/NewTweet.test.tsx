import React from 'react';
import { render, screen } from '@testing-library/react';
// @ts-ignore
import { create } from 'react-test-renderer';

import { NewTweet } from './NewTweet';

describe('NewTweet', () => {
  it('render NewTweet component', () => {
    render(<NewTweet/>);

    // Tweet button
    expect(screen.getByText('Tweet')).toBeInTheDocument();
    // textarea
    expect(screen.getByPlaceholderText('What\'s happening?')).toBeInTheDocument();
  });

  it('render NewTweet to snapshot', () => {
    const renderer = create(<NewTweet/>);
    expect(renderer.toJSON()).toMatchSnapshot();
  });
});
