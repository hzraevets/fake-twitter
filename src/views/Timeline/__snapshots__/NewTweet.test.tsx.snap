// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewTweet render NewTweet to snapshot 1`] = `
<div
  className="ant-card ant-card-bordered w-80 break-inside-avoid-column mb-4 css-dev-only-do-not-override-diro6f"
>
  <div
    className="ant-card-body"
    style={Object {}}
  >
    <div
      className="ant-space css-dev-only-do-not-override-diro6f ant-space-vertical w-full"
      style={Object {}}
    >
      <div
        className="ant-space-item"
        style={
          Object {
            "marginBottom": 16,
          }
        }
      >
        <textarea
          className="ant-input max-h-32 css-dev-only-do-not-override-diro6f"
          disabled={false}
          onChange={[Function]}
          onCompositionEnd={[Function]}
          onCompositionStart={[Function]}
          onKeyDown={[Function]}
          placeholder="What's happening?"
          rows={3}
          style={
            Object {
              "resize": undefined,
            }
          }
          value=""
        />
      </div>
      <div
        className="ant-space-item"
        style={Object {}}
      >
        <button
          className="ant-btn css-dev-only-do-not-override-diro6f ant-btn-primary float-right"
          disabled={true}
          onClick={[Function]}
          type="button"
        >
          <span>
            Tweet
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
